packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -473,7 +473,12 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
         return getWithRetryForTokenRefresh(async (options) => {
             const storageToken = await this.getStorageToken(options, "GetVersions");
-            const { url, headers } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/versions?count=${count}`, storageToken);
+            // If version is null, that means we need to fetch the oldest snapshot. For that use 50 for the number of
+            // snapshot versions to be fetched. The last snapshot version would be the oldest. ODSP doesn't store more
+            // than 50 snapshots and always retain the oldest snapshot. Also they don't allow requesting more than 50
+            // snapshots. This is required as we don't have any way to refer to oldest snapshot for now.
+            // TODO: Think of api to fetch oldest snapshot(server/client side) like fetch snapshot with seq number 0.
+            const { url, headers } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/versions?count=${blobid !== null ? count : 50}`, storageToken);
```

Comment:
I'm confused on 50. Where does it come from? I.e. why 50, not count? Caller is always in control of the count, no?

Reply:
I made this comment below, but i think using an op here might be a better solution. Basically we load from ops when null just like now, but the first op basically says you better have this summary or equivalent state to move forward from here. I like this as well as if in the future we truncate ops, we'd truncate to some summary op, and the same logic would still work.

Reply:
we might still want this at backcompat/short term, basically if we didn't load from summary, and the first op isn't summary or code proposal we need to go load some summary

Reply:
I think we can think of op implementation in near future as that would need server support too. They need to send that op and for that they should have the oldest snapshot version that they would send in the op. 
One other alternative is that to have a api in storage which allows fetchng snapshots or versions according to seq number. So in order to fetch the oldest snapshot we can fetch the snapshot with seq number 0. 
@tanviraumi Yes 50 is the max odsp allows in the query that we make, however they don't even store more than 10 now. As more snapshots keep coming in, they start purging snapshots, however they never purge the first one.

Reply:
maybe. couldn't we just stick and op in the op stream on attach as the first op

Reply:
regardless. if that is a path we think we want to go down. we should consider it for this change. so null would still mean load from ops, but we would look at the initial ops to determine if we need to load a snapshot, which is quite different than what we have here

Reply:
But I am bit skeptical about going on op path for loading for oldest snapshot because:
1.) We don't do such thing when we load from latest/specified snapshot.
2.) It would mean that we would use kind of null runtime and then look at ops and then switch context to actual runtime after fetching that snapshot. I don't think we want to do that given we moved to detached flow with one reason to get rid of it.

Reply:
1.) And i don't think we need to i see the link between the op and the summary and bi-directional, so you can start with the summary, and then you see the op that says you must have state equivalent to some sequence to proceed, and we would know that we do. for the null case, we would see the op and realize we didn't and have to load it.

2) It's true, but i don't think we will be able to get rid of it for the forceable future anyway. I also don't have a problem with it as an internal implementation detail, which this would allow.